### PR TITLE
LibraryTypeahead with demo endpoint

### PR DESCRIPTION
**Pull request by andrewconner**
@leogrim Opening for review, going to go ahead and make an endpoint we can play with this with.

### CODE REVIEW

**Comment by andrewconner** [comment]
Going to merge soon because of the big circular dependency work / merge conflicts. Reviews appreciated.

**Comment by ryanpbrewster** [comment]
I think that splitting out `OrgAvatarUploadCommander` and `BasicOrgGen` is super pleasant. :+1:

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationInfoCommander.scala [hunk 1]
```diff
@@ -40,15 +40,15 @@ trait OrganizationInfoCommander {
...
     orgMembershipRepo: OrganizationMembershipRepo,
-    orgInviteCommander: OrganizationInviteCommander,
-    organizationDomainOwnershipCommander: OrganizationDomainOwnershipCommander,
+    orgInviteCommander: Provider[OrganizationInviteCommander],
+    organizationDomainOwnershipCommander: Provider[OrganizationDomainOwnershipCommander],
     orgInviteRepo: OrganizationInviteRepo,
-    organizationAvatarCommander: OrganizationAvatarCommander,
-    slackInfoCommander: SlackInfoCommander,
+    organizationAvatarCommander: Provider[OrganizationAvatarCommander],
+    slackInfoCommander: Provider[SlackInfoCommander],
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationInfoCommander.scala:51]: Are these providers necessary now that you've cut some dependencies?

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationInfoCommander.scala:51]: Sorta. There were quite a few circular dependencies I found, not all fixed. Previously, Guice was auto-providering, which is way magic and randomly breaks like it did here. Could be some are okay now, but it took me 2 hours to pass tests after it started to break.

File: kifi-backend/modules/shoebox/conf/shoeboxService.routes [hunk 2]
```diff
@@ -130,6 +130,7 @@ POST    /ext/pref/email/:kind/:send @com.keepit.controllers.ext.ExtPreferenceCon
 
 GET     /ext/contacts/search        @com.keepit.controllers.ext.ExtUserController.searchForContacts(query:Option[String], limit:Option[Int])
 POST    /ext/contacts/hide          @com.keepit.controllers.ext.ExtNonUserSearchController.hideEmailFromUser()
+GET     /ext/keeps/suggest          @com.keepit.controllers.ext.ExtUserController.suggestRecipient(query:Option[String], limit:Option[Int])
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/conf/shoeboxService.routes]: This endpoint seems surprising. I would expect it to suggest keeps, not recipients.

Comment (andrewconner) [inline kifi-backend/modules/shoebox/conf/shoeboxService.routes]: I'll rename it. No clients are going to use it immediately.

**Comment by andrewconner** [comment]
If this breaks the build, I'll be sad.

**Pull request merged**